analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs

```diff
@@ -149,49 +133,45 @@ private static bool ParameterTypesDoNotMatch(IMethodSymbol method1, IMethodSymbo
 
         private static void ReportRedundantNullableConstructorCall(SyntaxNodeAnalysisContext context)
         {
-            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+            var objectCreation = ObjectCreationFactory.Create(context.Node);
             if (!IsNullableCreation(objectCreation, context.SemanticModel))
             {
                 return;
             }
 
-            if (IsInNotVarDeclaration(objectCreation) ||
-                IsInAssignmentOrReturnValue(objectCreation) ||
-                IsInArgumentAndCanBeChanged(objectCreation, context.SemanticModel))
+            if (IsInNotVarDeclaration(objectCreation.Expression)
+                || IsInAssignmentOrReturnValue(objectCreation.Expression)
+                || IsInArgumentAndCanBeChanged(objectCreation, context.SemanticModel))
             {
-                ReportIssueOnRedundantObjectCreation(context, objectCreation, "explicit nullable type creation", RedundancyType.ExplicitNullable);
+                ReportIssueOnRedundantObjectCreation(context, objectCreation.Expression, "explicit nullable type creation", RedundancyType.ExplicitNullable);
             }
         }
 
-        private static bool IsNullableCreation(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
+        private static bool IsNullableCreation(IObjectCreation objectCreation, SemanticModel semanticModel)
         {
-            if (objectCreation.ArgumentList == null ||
-                objectCreation.ArgumentList.Arguments.Count != 1)
+            if (!(objectCreation.ArgumentList is {Arguments: {Count: 1}}))
             {
                 return false;
             }
 
-            var type = semanticModel.GetSymbolInfo(objectCreation).Symbol?.ContainingType;
-            return type != null &&
-                type.OriginalDefinition.Is(KnownType.System_Nullable_T);
+            var type = semanticModel.GetSymbolInfo(objectCreation.Expression).Symbol?.ContainingType;
+            return type != null && type.OriginalDefinition.Is(KnownType.System_Nullable_T);
         }
 
-
-        private static bool IsInAssignmentOrReturnValue(ObjectCreationExpressionSyntax objectCreation)
+        private static bool IsInAssignmentOrReturnValue(SyntaxNode objectCreation)
         {
             var parent = objectCreation.GetFirstNonParenthesizedParent();
-            return parent is AssignmentExpressionSyntax ||
-                parent is ReturnStatementSyntax ||
-                parent is LambdaExpressionSyntax;
+            return parent is AssignmentExpressionSyntax
+                   || parent is ReturnStatementSyntax
+                   || parent is LambdaExpressionSyntax;
         }
 
-        private static bool IsInNotVarDeclaration(ObjectCreationExpressionSyntax objectCreation)
+        private static bool IsInNotVarDeclaration(SyntaxNode objectCreation)
```

Comment:
We need a C# 9 specific compliant test case that will enter this section and find variable declaration, because it's using the `.Parent.Parent.Parent`. It should work if I did the finger counting correctly, but let's be safe.